service/javadsl/integration-client/src/main/java/com/lightbend/lagom/javadsl/client/integration/LagomClientFactory.java

```diff
@@ -167,10 +168,25 @@ private LagomClientFactory(EventLoopGroup eventLoop, WSClient wsClient, WebSocke
     public void close() {
         closeGracefully(wsClient::close);
         closeGracefully(webSocketClient::shutdown);
```

Comment:
I think the correct way to handle resources in this class would be to register tasks on coordinated shutdown as soon as the resource is allocated (if allocated here) and then simplify `close()`:

```
public void close() {
    coordinatedShutdown(); 
}
```

I think with this approach coordinated shutdown takes over completely and `closeGracefully` is no longer necessary. 

(haven't looked into it in detail though, there may be edge cases I'm missing)

Reply:
WDYT?